pallets/fragments/src/lib.rs

```diff
@@ -1066,256 +1106,227 @@ pub mod pallet {
-				Error::<T>::NoPermission
+			// create an account for a specific fragment
+			// we need an existential amount deposit to be able to create the vault account
+			let frag_account = Self::get_fragment_account_id(definition_hash, edition, copy);
+			let minimum_balance =
+				<pallet_balances::Pallet<T> as fungible::Inspect<T::AccountId>>::minimum_balance();
+			let _ = <pallet_balances::Pallet<T> as fungible::Mutate<T::AccountId>>::mint_into(
+				&frag_account,
+				minimum_balance,
 			);
 
-			let perms = if let Some(new_perms) = new_permissions {
-				// ensure we only allow more restrictive permissions
-				if (item_data.permissions & FragmentPerms::EDIT) != FragmentPerms::EDIT {
-					ensure!(
-						(new_perms & FragmentPerms::EDIT) != FragmentPerms::EDIT,
-						Error::<T>::NoPermission
-					);
-				}
-				if (item_data.permissions & FragmentPerms::COPY) != FragmentPerms::COPY {
-					ensure!(
-						(new_perms & FragmentPerms::COPY) != FragmentPerms::COPY,
-						Error::<T>::NoPermission
-					);
-				}
-				if (item_data.permissions & FragmentPerms::TRANSFER) != FragmentPerms::TRANSFER {
-					ensure!(
-						(new_perms & FragmentPerms::TRANSFER) != FragmentPerms::TRANSFER,
-						Error::<T>::NoPermission
-					);
-				}
-				new_perms
-			} else {
-				item_data.permissions
-			};
+			// TODO Make owner pay for deposit actually!
+			// TODO setup proxy
 
-			let to = T::Lookup::lookup(to)?;
+			Ok(())
+		}
 
-			// now we take two different paths if item can be copied or not
-			if (item_data.permissions & FragmentPerms::COPY) == FragmentPerms::COPY {
-				// we will copy the item to the new account
-				item_data.permissions = perms;
 
-				let copy: u64 = <CopiesCount<T>>::get((definition_hash, edition))
-					.ok_or(Error::<T>::NotFound)?
-					.into();
+		/// Put a Fragment Instance on sale.
+		///
+		///
+		/// Note: **Only** the **Fragment Instance's owner** is **allowed** to call this extrinsic
+		///
+		/// # Arguments
+		///
+		/// * `origin` - **Origin** of the **extrinsic function**
+		/// * `definition_hash` - Fragment Definition ID of the Fragment Instance
+		/// * `edition` - Edition ID of the Fragment Instance
+		/// * `copy` - Copy ID of the Fragment instance
+		/// * `to` - **Account ID** to give the Fragment instance
+		/// * `new_permissions` (*optional*) - The permitted set of actions that the account that is given the Fragment instance can do with it. Note: `new_permissions` must be a subset of the current `permissions` field of the Fragment Instance.
+		/// * `expiration` (*optional*) - Block number that the newly-copied Fragment Instance expires at. If the Fragment Instance is not copyable, this parameter is practically irrelevant.
+		/// * `secondary_sale_type` - Type of Sale
+		#[pallet::weight(50_000)]
+		pub fn resell(
+			origin: OriginFor<T>,
+			definition_hash: Hash128,
+			edition_id: Unit,
+			copy_id: Unit,
+			new_permissions: Option<FragmentPerms>,
```

Comment:
are we checking that new perms are compatible with existing ones and more restrictive or equal?

Reply:
of course. that's we we call `Self::can_transfer_instance` in this extrinsic